src/functions/types.ts

```diff
@@ -131,11 +131,18 @@ type ErroredFunctionReturnArgs<OutputParameters> =
   & Partial<SuccessfulFunctionReturnArgs<OutputParameters>>
   & Required<Pick<SuccessfulFunctionReturnArgs<OutputParameters>, "error">>;
 
+type PendingFunctionReturnArgs<OutputParameters> =
+  & {
+    completed: false;
+  }
+  & Partial<SuccessfulFunctionReturnArgs<OutputParameters>>;
```

Comment:
@selfcontained do we want this partial here or should we just allow `{ completed: false }`?

Reply:
Just the completed: false - If a functions is return that, it indicates they’ll be calling an api to complete the function at a later time, and provide their outputs then. Providing outputs when returning completed: false wouldn’t have any effect with the outputs, so I don’t think we should signal they can return outputs too in that situation.